CryptoTechReminderSystem.Test/Gateway/SlackGatewayTests.cs

```diff
@@ -22,18 +22,64 @@ public class SlackGatewayTests
         [TestFixture]
         public class CanSendMessage
         {
-            private FluentSimulator _slackApi;
+            private SlackSimulator _slackApi;
             private SlackGateway _slackGateway;
             
+            class SlackSimulator
+            {
+                private FluentSimulator _simulator;
+                public bool SpyMethodCalled;
+                public string ReceivedErrorMessage;
+
+                public SlackSimulator(string address)
+                {
+                    _simulator = new FluentSimulator(address);
+                    SpyMethodCalled = false;
+                }
+                
+                public ReceivedRequest[] ReceivedRequests()
+                {
+                    return _simulator.ReceivedRequests.ToArray();
+                }
+
+                public void ShouldRespondWithOk()
+                {
+                    _simulator.Post("/" + PostMessageApiPath).Responds(new { ok = true });
+                }
+                public void ShouldRespondWithError()
+                {
+                    _simulator.Post("/" + PostMessageApiPath).Responds(
+                        new {ok = false, error = "error message"});
+                }
+              
+                public void Start()
+                {
+                    _simulator.Start();
+                }
+
+                public void Stop()
+                {
+                    _simulator.Stop();
+                }
+
+                public void SpyMethod()
+                {
+                    SpyMethodCalled = true;
+                }
+
+                public void HandleError(Exception ex)
```

Comment:
Instead of `Exception ex`, it could be `Exception exception` for readability.

Reply:
Resolved in https://github.com/chubberlisk/crypto_tech/pull/37/commits/ec68aefbafc142b8995bd1f01c6b42f237c70c8e